### DISCUSSION TITLE
Minimal DI Registration w/o AssemblyScanning

### DISCUSSION THREAD

**Discussion by TorreyGarland** [LuckyPennySoftware/MediatR#880] [Q&A] [answered]
For a .NET Core project that I am working on, we are using Mediatr, but we have our own mechanism for registering dependencies using SourceGenerators.

What is the minimum registrations to get Mediatr working without using the Mediatr DI package?

**Comment by jbogard** [comment, marked as answer]
I'm not sure what your source generators do, but you can see the difference between what you have and this: https://github.com/jbogard/MediatR/blob/master/src/MediatR/Registration/ServiceRegistrar.cs

**Reply by TorreyGarland** [reply]
Thanks for the quick response and point in the right direction!
